Duplicati/Library/Backend/TencentCOS/COSBackend.cs

```diff
@@ -0,0 +1,421 @@
+﻿using COSXML;
+using COSXML.Auth;
+using COSXML.Model.Bucket;
+using COSXML.Model.Object;
+using COSXML.Model.Tag;
+using COSXML.Utils;
+using Duplicati.Library.Common.IO;
+using Duplicati.Library.Interface;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Duplicati.Library.Backend
+{
+    /// <summary>
+    /// Tencent COS
+    /// https://cloud.tencent.com/document/product/436
+    /// https://cloud.tencent.com/document/product/436/32869
+    /// </summary>
+    public class COS : IBackend, IStreamingBackend, IRenameEnabledBackend
+    {
+        private static readonly string LOGTAG = Logging.Log.LogTagFromType<COS>();
+
+        private const string COS_APP_ID = "cos-app-id";
+        private const string COS_REGION = "cos-region";
+        private const string COS_SECRET_ID = "cos-secret-id";
+        private const string COS_SECRET_KEY = "cos-secret-key";
+        private const string COS_BUCKET = "cos-bucket";
+
+        private const bool COS_USE_SSL = true;
+        private const int CONNECTION_TIMEOUT_MS = 1000 * 60 * 60;
+        private const int READ_WRITE_TIMEOUT_MS = 1000 * 60 * 60;
+        private const int KEY_DURATION_SECOND = 60 * 60;
+
+        private readonly CosOptions _cosOptions;
+        private readonly string m_prefix;
+
+        private CosXml _cosXml;
+
+        /// <summary>
+        /// 腾讯云对象存储配置
```

Comment:
Can you translate all the Chinese comments into English?  This would make it easier for other developers to understand the code.